drivers/clock_control/nrf_power_clock.c

```diff
@@ -499,38 +469,103 @@ void z_nrf_clock_control_lf_on(enum nrf_lfclk_start_mode start_mode)
 	}
 }
 
-/* Note: this function has public linkage, and MUST have this
- * particular name.  The platform architecture itself doesn't care,
- * but there is a test (tests/kernel/arm_irq_vector_table) that needs
- * to find it to it can set it in a custom vector table.  Should
- * probably better abstract that at some point (e.g. query and reset
- * it by pointer at runtime, maybe?) so we don't have this leaky
- * symbol.
- */
-void nrf_power_clock_isr(void *arg);
+static void clock_event_handler(nrfx_clock_evt_type_t event)
+{
+	struct device *dev = DEVICE_GET(clock_nrf);
+	switch (event) {
+	case NRFX_CLOCK_EVT_HFCLK_STARTED:
+	{
+		struct nrf_clock_control_sub_data *data =
+				get_sub_data(dev, CLOCK_CONTROL_NRF_TYPE_HFCLK);
+
+		/* Check needed due to anomaly 201:
+		 * HFCLKSTARTED may be generated twice.
+		 */
+		if (GET_STATUS(data->flags) == CLOCK_CONTROL_STATUS_STARTING) {
+			clkstarted_handle(dev, CLOCK_CONTROL_NRF_TYPE_HFCLK);
+		}
+
+		break;
+	}
+	case NRFX_CLOCK_EVT_LFCLK_STARTED:
+		if (IS_ENABLED(
+			CONFIG_CLOCK_CONTROL_NRF_K32SRC_RC_CALIBRATION)) {
+			z_nrf_clock_calibration_lfclk_started();
+		}
+		clkstarted_handle(dev, CLOCK_CONTROL_NRF_TYPE_LFCLK);
+		break;
+	case NRFX_CLOCK_EVT_CAL_DONE:
+		if (IS_ENABLED(CONFIG_CLOCK_CONTROL_NRF_K32SRC_RC_CALIBRATION)) {
+			z_nrf_clock_calibration_done_handler();
+		}
+		break;
+	default:
+		__ASSERT_NO_MSG(0);
+		break;
+	}
+}
+
+#if defined(CONFIG_USB_NRFX)
```

Comment:
Is there a reason why those USB related things are still kept here? It seems that they all could be moved to `usb_dc_nrfx.c`.

Reply:
no reason, moved it to `usb_dc_nrfx.c`. Only left `USBREGULATOR_IRQn` interrupt connection because it is using same priority as clock power interrupt.

Reply:
I think it should be moved as well. It's purely USB related stuff. And you can get the power/clock IRQ priority also from `usb_dc_nrfx.c`, possibly with a comment why this priority is used.

Reply:
done